datasciencecampus/transport-network-performance
# 99 select layer

## Description
- Added functionality to define band when loading raster file in `transport_peformance.population.rasterpop`.
- Added defences to test expected behaviour of the argument.
- Added relevant unit tests.
- Modified docstrings.

Fixes #99 

## Motivation and Context
GHSL raster files only include a single band. However, if using different data sources, a functionality may be needed to select a specific band when loading the data (e.g. cases where the population data is in a band other than the first).

## Type of change

## How Has This Been Tested?
- Run code snippets in isolation to test the behaviour of the `xarray` `sel` method.
- Run relevant functions with a dummy raster file with more than one band.
- Run relevant parts of the `e2e.py` notebook to ensure pipeline works.
- Run unit tests locally.

Test configuration details:
* OS: macOS Ventura 13.6
* Python version: 3.9.13
* Java version: n/a
* Python management system: conda/pip

## Advice for reviewer


## Checklist:

- [x] My code follows the intended structure of this project

## Additional comments
Made three small changes unrelated to the issue:
- Small typo in `test_rasterpop_get_pop_raises` docstring (`get_data()` -> `get_pop()`).
- Small typo in `test_plot_folium_no_uc` function name.
- Added `centre_crs` argument to urban centre module call in `e2e.py` notebook.
